test/helpers/compile_deploy.js

```diff
@@ -0,0 +1,46 @@
+const { readFileSync } = require("fs");
+const { compile } = require("solc");
+
+/**
+ * Compile and deploy the selected contract(s)
+ * @param {String} contractPath  Path to contracts directory
+ * @param {String} mainContractName  Name of the main contract (without .sol extension)
+ * @param {Array|String} contractFileNames List of imported contracts
+ * @param {Object} web3 Web3 interface
+ * @returns {Object} context
+ */
+async function compileAndDeploy(contractPath, mainContractName, contractFileNames = [], web3) {
+  const selectedContracts = contractFileNames.length === 0 ? [mainContractName] : contractFileNames;
+  const contractSources = selectedContracts.map((contractName) => {
+    return { [`${contractName}.sol`]: readFileSync(`${contractPath}${contractName}.sol`, "utf8") };
+  });
+  const sources = Object.assign(...contractSources);
```

Comment:
I'm not sure we need this line here since this is the last usage of `contractSources`.  If we do need a copy (in case `compile` mutates `sources` on line 19) I would change this to explicitly use an empty object as the source parameter `Object.assign({}, contractSources)` or just the spread operator like `const sources = {...contractSources}` and use the original `contractSources` in the return on line 40.

Reply:
Agreed. Wanted to keep the code readable.